client/state/jetpack-connect/reducer.js

```diff
@@ -84,7 +89,7 @@ export function jetpackConnectAuthorize( state = {}, action ) {
 			return Object.assign( {}, state, { siteReceived: true, isAuthorizing: false } );
 		case JETPACK_CONNECT_QUERY_SET:
 			const queryObject = Object.assign( {}, action.queryObject );
-			return Object.assign( {}, defaultAuthorizeState, { queryObject: queryObject } );
+			return Object.assign( {}, state, defaultAuthorizeState, { queryObject: queryObject } );
```

Comment:
For Jetpack SSO, the flow roughly looks like.

- User lands on `/jetpack/sso`.
- We validate their nonces, etc.
- Then user can click a "Log in" button
- If user clicks that, we fire off the `JETPACK_CONNECT_SSO_AUTHORIZE` action which sets `autoAuthorize` to true
- Then, if everything works, we redirect user to wp-admin
- After wp-admin, user gets redirect to `/jetpack/connect/authorize` where [JETPACK_CONNECT_QUERY_SET is called](https://github.com/Automattic/wp-calypso/blob/master/client/signup/jetpack-connect/controller.js#L30].
- If we removed `state` as suggested, then the `autoAuthorize` value that we set on `JETPACK_CONNECT_SSO_AUTHORIZE` will be overridden as well.

Perhaps, there's another way to address this, but we need to find a way to persist that `autoAuthorize` property.